api/v1/api.go

```diff
@@ -77,18 +81,160 @@ type postIsiExportResp struct {
 type IsiExport struct {
 	ID    int      `json:"ID"`
 	Paths []string `json:"paths"`
+	Clients []string `json:"clients"`
 }
 
 type getIsiExportsResp struct {
 	ExportList []*IsiExport `json:"exports"`
 }
 
+type isiThresholds struct {
+	Advisory int64 `json:"advisory"`
+	AdvisoryExceeded bool `json:"advisory_exceeded"`
+	AdvisoryLastExceeded interface{} `json:"advisory_last_exceeded"`
+	Hard int64 `json:"hard"`
+	HardExceeded bool `json:"hard_exceeded"`
+	HardLastExceeded interface{} `json:"hard_last_exceeded"`
+	Soft int64 `json:"soft"`
+	SoftExceeded bool `json:"soft_exceeded"`
+	SoftLastExceeded interface{} `json:"soft_last_exceeded"`
+}
+
+type IsiQuota struct {
+	Container bool `json:"container"`
+	Enforced  bool `json:"enforced"`
+	Id        string `json:"id"`
+	IncludeSnapshots bool `json:"include_snapshots"`
+	Linked interface{} `json:"linked"`
+	Notifications string `json:"notifications"`
+	Path string `json:"path"`
+	Persona interface{} `json:"persona"`
+	Ready bool `json:"ready"`
+	Thresholds isiThresholds `json:"thresholds"`
+	ThresholdsIncludeOverhead bool `json:"thresholds_include_overhead"`
+	Type string `json:"type"`
+	Usage struct {
+		Inodes int64 `json:"inodes"`
+		Logical int64 `json:"logical"`
+		Physical int64 `json:"physical"`
+	} `json:"usage"`
+}
+
+type isiThresholdsReq struct {
+	Advisory interface{} `json:"advisory"`
+	Hard interface{} `json:"hard"`
+	Soft interface{} `json:"soft"`
+}
+
+type IsiQuotaReq struct {
+	Enforced  bool `json:"enforced"`
+	IncludeSnapshots bool `json:"include_snapshots"`
+	Path string `json:"path"`
+	Thresholds isiThresholdsReq `json:"thresholds"`
+	ThresholdsIncludeOverhead bool `json:"thresholds_include_overhead"`
+	Type string `json:"type"`
+}
+
+type IsiUpdateQuotaReq struct {
+	Enforced  bool `json:"enforced"`
+	Thresholds isiThresholdsReq `json:"thresholds"`
+	ThresholdsIncludeOverhead bool `json:"thresholds_include_overhead"`
+}
+
+type isiQuotaListResp struct {
+	Quotas []IsiQuota `json:"quotas"`
+}
+
+// Get the quota for a directory
+func (papi *PapiConnection) GetIsiQuota(path string) (quota *IsiQuota, err error) {
+	// PAPI call: GET https://1.2.3.4:8080/platform/1/quota/quotas
+	// This will list out all quotas on the cluster
+	
+	var quotaResp isiQuotaListResp
+	err = papi.query("GET", papiQuotaPath, "", nil, nil, &quotaResp)
+	if err != nil {
+		return nil, err
+	}
+	
+	// find the specific quota we are looking for
+	for _, quota := range quotaResp.Quotas {
+		if quota.Path == path {
+			return &quota, nil
+		}
+	}
+	
+	return nil, errors.New(fmt.Sprintf("Quota not found: %s", path))
+}
+
+// Set the quota for a directory
+func (papi *PapiConnection) SetIsiQuota(path string, size int64) (err error) {
```

Comment:
So because goisilon is a separate package from REX-Ray, I feel like the functions it implements should not be limited to the capabilities required by REX-Ray. So for example, this function should not have hard coded parameters for the IsiQuotaReq, they should be arguments to the function, possibly with default values to make them optional. Or another option would be to have the SetIsiQuota function's input be a reference to a IsiQuotaReq that the caller configures prior to calling SetIsiQuota or if GO supports function overloading you could do both. That way clients aren't limited to the needs of REX-Ray should someone decide to use this outside of REX-Ray.